engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_file_stream_logs_strategy.go

```diff
@@ -0,0 +1,124 @@
+package stream_logs_strategy
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_consts"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_filesystem"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/logline"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"io"
+	"strings"
+)
+
+// This strategy pulls logs from filesytsem where there is a log file per enclave, per service
+type PerFileStreamLogsStrategy struct {
+}
+
+func NewPerFileStreamLogsStrategy() *PerFileStreamLogsStrategy {
+	return &PerFileStreamLogsStrategy{}
+}
+
+type JsonLog map[string]string
+
+func (strategy *PerFileStreamLogsStrategy) StreamLogs(
+	ctx context.Context,
+	fs volume_filesystem.VolumeFilesystem,
+	logsByKurtosisUserServiceUuidChan chan map[service.ServiceUUID][]logline.LogLine,
+	streamErrChan chan error,
+	enclaveUuid enclave.EnclaveUUID,
+	serviceUuid service.ServiceUUID,
+	conjunctiveLogLinesFiltersWithRegex []logline.LogLineFilterWithRegex,
+	shouldFollowLogs bool,
+) {
+	// logs are stored per enclave id, per service uuid, eg. <base path>/123440231421/54325342w2341.json
+	logsFilepath := fmt.Sprintf("%s%s/%s%s", volume_consts.LogsStorageDirpath, string(enclaveUuid), string(serviceUuid), volume_consts.Filetype)
```

Comment:
have a constant for this one as well maybe. live computing those log file paths all over the place is suuuper fragile and worrying me a little TBH